internal/state/execution.go

```diff
@@ -466,9 +466,10 @@ func buildExtendedCommitInfo(ec *types.ExtendedCommit, store Store, initialHeigh
 		}
 
 		var ext []byte
-		if ecs.BlockIDFlag == types.BlockIDFlagCommit {
-			// We only care about vote extensions if a validator has voted to
-			// commit.
+		if ap.VoteExtensionsEnabled(ec.Height) {
```

Comment:
Good question, and I got a bit tripped up here. The reason that this works is because we use the _commit_ height here, `ec.Height`. This check is asking the question "For that height that produced `ec`, were extensions required for consensus to proceed"? We get around the off-by-one confusion by phrasing the question that way. Obviously, the first proposal height to receive extensions will be where the answer to that question is "yes" from the previous height, but I find having to think about that a bit more confusing so I used `ec.Height` here. 

The new test in `execution_test.go`, https://github.com/tendermint/tendermint/pull/8547/files#diff-df92b7c0a40234177a86250715d10842bcbc0e7275f347c68a517c135bbdedddR1010, should hopefully also help convince you that this is right.

Reply:
commented.

Reply:
I saw, thanks a lot!